casper/src/main/resources/PoS.rhox

```diff
@@ -160,30 +201,119 @@ new  PoS,
             } |
 
             contract PoS(@"closeBlock", ackCh) = {
-              new blockDataCh, mvarProcessCh,
-                  getBlockData(`rho:block:data`),
-                  commitReward, rewardsCh, newValidatorsCh in {
-                getBlockData!(*blockDataCh) |
+              new commitRewards,
+                  changeEpoch,
+                  removeQuarantinedWithdrawers
+              in {
+                new blockDataCh, mvarProcessCh,
+                    getBlockData(`rho:block:data`),
+                    newRewardsCh, newValidatorsCh,
+                    newWithdrawersCh, newBondsCh
+                in {
+                  getBlockData!(*blockDataCh) |
+
+                  runMVar!(*stateCh, *mvarProcessCh, *ackCh) |
+                  for(_, @blockNumber <- blockDataCh;
+                      @state, mvarResultCh <- mvarProcessCh) {
+                    changeEpoch!(
+                      blockNumber,
+                      state.get("allBonds"),
+                      state.get("activeValidators"),
+                      state.get("withdrawers"),
+                      *newValidatorsCh)|
+
+                    removeQuarantinedWithdrawers!(
+                      blockNumber,
+                      state.get("allBonds"),
+                      state.get("withdrawers"),
+                      *newWithdrawersCh,
+                      *newBondsCh) |
```

Comment:
So I see this "method" gives back the new withdraw and new bond set but where does the withdrawals actually go to?

Reply:
For now, they just go to a better place :)

Now I realise that they should receive their bonds and their rewards. I have to  fix this in this PR

Reply:
fixed